src/Admin/AbstractAdmin.php

```diff
@@ -2712,6 +2712,11 @@ public function hasAccess($action, $object = null)
      */
     public function configureActionButtons($action, $object = null)
     {
+        // nothing to do for non-internal actions
+        if (!\in_array($action, ['tree', 'show', 'edit', 'delete', 'acl', 'history', 'list', 'batch'], true)) {
+            return [];
+        }
```

Comment:
All the following code for this method does not make sense for custom actions. Perhaps it would be better to immediately identify these custom actions?

Reply:
indeed

Reply:
Can someone explain to me why this is the case? I have custom actions from where I'd like to be able to move back to "edit" mode.

Reply:
there is a `configureActionButtons` method for this.

Reply:
Oh, you meant the early return ?

All the logic here doesn't make sens for custom action, the call to `configureActionButtons` still does. Can you open a pr @toooni ?

Reply:
Yes, the early return. Yes, the configureActionButtons call does always make sense and I can make a PR. But why does an edit action button not make sense in a custom action? I'd think the default behaviour should be that the the edit (and other actions) should be visible like in internal actions (these could still be unset in `configureActionButtons`).

Reply:
I recommend you to open the PR with what you think is the good behavior and we'll discuss about it.

But why does an edit action button would make sens in a custom action too ? For each actions, we have a check like
```
self::MASK_OF_ACTION_EDIT & $actionBit
```
Removing the early return won't add automatically the edit button in your custom action, and I think it's the right move. Custom action are custom, so buttons should be configured.

Reply:
It has always been a challenge to add standard action buttons to custom action pages. But this must always be manually configured. Not every custom page needs to add an edit button.

Reply:
Ok. I might be a bit biased because in my cases I could always need the edit action button. I've created a PR https://github.com/sonata-project/SonataAdminBundle/pull/7483 to allow `configureActionButtons` again.

Btw. maybe the default action buttons should be predefined somewhere so it would be easier to re-add them in custom actions.